website_sale_affiliate/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+import logging
+
+from odoo.http import request, route
+from odoo.addons.website_sale.controllers.main import WebsiteSale as Base
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(Base):
+    def _store_affiliate_info(self, **kwargs):
```

Comment:
@hughesbm Keep in mind no rule is a hard rule - they can all be broken with a good reason